mathesar/imports/json.py

```diff
@@ -1,24 +1,47 @@
 import json
+from json.decoder import JSONDecodeError
 
 from db.tables.operations.alter import update_pk_sequence_to_latest
 from mathesar.database.base import create_mathesar_engine
 from db.records.operations.insert import insert_records_from_json
 from db.tables.operations.create import create_string_column_table
 from db.tables.operations.drop import drop_table
+from mathesar.api.exceptions.database_exceptions import (
+    exceptions as database_api_exceptions
+)
 from mathesar.imports.utils import process_column_names
 from db.constants import ID, ID_ORIGINAL
 from psycopg2.errors import IntegrityError, DataError
 
 from mathesar.state import reset_reflection
 
 
+def validate_json_format(data_file_content):
+    try:
+        data = json.load(data_file_content)
+    except (JSONDecodeError, ValueError) as e:
+        raise database_api_exceptions.InvalidJSONFormat(e)
+
+    if isinstance(data, list) and all(isinstance(val, dict) for val in data):
```

Comment:
A variable name to communicate intent of this boolean composition would be nice.

```suggestion
    is_list_of_dicts = isinstance(data, list) and all(isinstance(val, dict) for val in data)
    if is_list_of_dicts:
```

Reply:
Thanks Dom, added it now.